simulation/traffic_simulator/src/api/api.cpp

```diff
@@ -397,4 +425,52 @@ void API::requestLaneChange(
   entity_manager_ptr_->requestLaneChange(name, target, trajectory_shape, constraint);
 }
 
+void API::requestSpeedChange(const std::string & name, double target_speed, bool continuous)
+{
+  if (entity_manager_ptr_ == nullptr) {
+    THROW_SIMULATION_ERROR(__FILE__, __LINE__, "Entity manager is not initialized");
+  }
+
+  entity_manager_ptr_->requestSpeedChange(name, target_speed, continuous);
+  if (entity_manager_ptr_->isEgo(name) and not isNpcLogicStarted()) {
+    ego_entity_simulation_->requestSpeedChange(target_speed);
+  }
+}
+
+void API::requestSpeedChange(
+  const std::string & name, const double target_speed, const speed_change::Transition transition,
+  const speed_change::Constraint constraint, const bool continuous)
+{
+  if (entity_manager_ptr_ == nullptr) {
+    THROW_SIMULATION_ERROR(__FILE__, __LINE__, "Entity manager is not initialized");
+  }
+
+  entity_manager_ptr_->requestSpeedChange(name, target_speed, transition, constraint, continuous);
+  if (entity_manager_ptr_->isEgo(name) and not isNpcLogicStarted()) {
+    ego_entity_simulation_->requestSpeedChange(target_speed);
+  }
+}
+
+void API::requestSpeedChange(
+  const std::string & name, const speed_change::RelativeTargetSpeed & target_speed, bool continuous)
+{
+  if (entity_manager_ptr_ == nullptr) {
+    THROW_SIMULATION_ERROR(__FILE__, __LINE__, "Entity manager is not initialized");
```

Comment:
THROW_SIMULATION_ERROR macro includes file and line infomation.

```suggestion
    THROW_SIMULATION_ERROR("Entity manager is not initialized");
```

Reply:
Also, entity_manager_ptr_ is initialized in member initialized list, so I think this error means the simulator was broken.
So please leave message such as "If you see this error, something completely unexpected happens. Please contact developer."